src/openrct2/ride/Ride.cpp

```diff
@@ -835,11 +836,8 @@ void Ride::FormatStatusTo(Formatter& ft) const
     {
         ft.Add<rct_string_id>(STR_TEST_RUN);
     }
-    else if (
-        mode == RIDE_MODE_RACE && !(lifecycle_flags & RIDE_LIFECYCLE_PASS_STATION_NO_STOPPING)
-        && race_winner != SPRITE_INDEX_NULL)
+    else if (mode == RIDE_MODE_RACE && !(lifecycle_flags & RIDE_LIFECYCLE_PASS_STATION_NO_STOPPING) && peep != nullptr)
```

Comment:
There is no need for this check as well as the one lower down. I would suggest removing this one and moving the `auto peep = GetEntity<Peep>(race_winner);` into L842 where it was before.

Reply:
I've refactored code as you suggested.